src/language/providers/completionProvider.ts

```diff
@@ -0,0 +1,187 @@
+import { CompletionItem, CompletionItemKind, languages } from "vscode";
+import Database, { SQLType } from "../../database/schemas";
+import Table from "../../database/table";
+import Document from "../sql/document";
+import { ObjectRef } from "../sql/types";
+import { updateCache } from "./completionItemCache";
+import CompletionItemCache from "./completionItemCache";
+
+const completionItemCache = new CompletionItemCache();
+
+export interface CompletionType {
+  order: string;
+  label: string;
+  type: SQLType;
+  icon: CompletionItemKind;
+}
+
+const completionTypes: { [index: string]: CompletionType } = {
+  tables: {
+    order: `a`,
+    label: `table`,
+    type: `tables`,
+    icon: CompletionItemKind.File,
+  },
+  views: {
+    order: `b`,
+    label: `view`,
+    type: `views`,
+    icon: CompletionItemKind.Interface,
+  },
+  aliases: {
+    order: `c`,
+    label: `alias`,
+    type: `aliases`,
+    icon: CompletionItemKind.Reference,
+  },
+};
+
+function createCompletionItem(
+  name: string,
+  kind: CompletionItemKind,
+  detail?: string,
+  documentation?: string,
+  sortText?: string
+): CompletionItem {
+  const item = new CompletionItem(name, kind);
+  item.detail = detail;
+  item.documentation = documentation;
+  item.sortText = sortText;
+  return item;
+}
+
+function getColumnAtributes(column: TableColumn): string {
+  return `Field: ${column.COLUMN_NAME}\n Type: ${column.DATA_TYPE}\n HAS_DEFAULT: ${column.HAS_DEFAULT}\n IS_IDENTITY: ${column.IS_IDENTITY}\n IS_NULLABLE: ${column.IS_NULLABLE}`;
+}
+
+async function getTableItems(
+  schema: string,
+  name: string
+): Promise<CompletionItem[]> {
+  if (!completionItemCache.has(schema + name)) {
+    const items = await Table.getItems(schema, name);
+    const completionItems = items.map((i) =>
+      createCompletionItem(
+        i.COLUMN_NAME,
+        CompletionItemKind.Field,
+        getColumnAtributes(i),
+        `Schema: ${schema}\nTable: ${name}\n`
+      )
+    );
+    completionItemCache.set(schema + name, completionItems);
+  }
+  return completionItemCache.get(schema + name);
+}
+
+async function getObjectCompletions(
+  curSchema: string,
+  sqlTypes: { [index: string]: CompletionType }
+) {
+  const schemaUpdate: boolean = updateCache.delete(curSchema);
+  if (!completionItemCache.has(curSchema) || schemaUpdate) {
+    const promises = Object.entries(sqlTypes).map(async ([_, value]) => {
+      const data = await Database.getObjects(curSchema, value.type);
+      return data.map((table) =>
+        createCompletionItem(
+          table.name,
+          value.icon,
+          `Type: ${value.label}`,
+          `Schema: ${table.schema}`,
+          value.order
+        )
+      );
+    });
+
+    const results = await Promise.all(promises);
```

Comment:
Let's change `Promise.all` to use `Promise.allSettled` as that will handle any unfortunate crashes that may occur.

Be sure to rewrite the flat logic beloe because `allSettled` has a different signiture.